doc/articles/contributing/build-troubleshooting.md

```diff
@@ -0,0 +1,51 @@
+# build troubleshooting
+
+
+## on your computer
+
+Start by deleting `mono-wasm-xxx` in `C:\Users\userid\AppData\Local\Temp\`.
+
+Increase the build log verbosity in [Visual Studio to diagnostics](https://docs.microsoft.com/en-us/visualstudio/msbuild/obtaining-build-logs-with-msbuild?view=vs-2019). 
+
+Head to Tools -> Options -> MS Build Verbosity
+
+- Change to detailed (window/console logger)
+- Change to detailed (log)
+
+Quit visual studio and ensure that all processes have been terminated.
+
+```bash
+taskkill /im "devenv*" /f /t
+taskkill /im "msbuild*" /f /t
+taskkill /im "uno*" /f /t
+```
+
+Reset the source generation cache and all local changes to your git repository. Be careful, as this will wipe out code you haven't yet commited.
+
+```bash
+git reset --hard
+git clean -fdx
+git checkout master
+git pull origin master
+```
+
+Open the solution again and wait for background tasks (bottom left blue corner) to be {} aka None aka Ready.
+
+Compile the `Uno UI -> Uno.UI` project.
+
+If that doesn't work, try the following.
+
+```
+cd src
+msbuild Uno.UI.sln /m /r /bl:restore.binlog /t:restore
+msbuild Uno.UI.sln /m /r /bl:build.binlog
+nuget locals all -clear
```

Comment:
Doing it in his order may erase evidence of cache errors. The clear nuget should be the first line.